src/main.ts

```diff
@@ -99,61 +102,61 @@ export class Minifier {
   private checkReserved(str: string): boolean {
     // From MDN's Lexical Grammar page
     // (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar)
-    var reserved = [
-      'break',
-      'case',
-      'class',
-      'catch',
-      'const',
-      'continue',
-      'debugger',
-      'delete',
-      'do',
-      'else',
-      'export',
-      'extends',
-      'finally',
-      'for',
-      'function',
-      'if',
-      'import',
-      'in',
-      'instanceof',
-      'let',
-      'new',
-      'return',
-      'super',
-      'switch',
-      'this',
-      'throw',
-      'try',
-      'typeof',
-      'var',
-      'void',
-      'while',
-      'with',
-      'yield',
-      'enum',
-      'await',
-      'int',
-      'byte',
-      'char',
-      'goto',
-      'long',
-      'final',
-      'float',
-      'short',
-      'double',
-      'native',
-      'throws',
-      'boolean',
-      'abstract',
-      'volatile',
-      'transient',
-      'synchronized'
-    ];
+    var reserved = {
+      'break' : true,
+      'case': true,
+      'class': true,
+      'catch': true,
+      'const': true,
+      'continue': true,
+      'debugger': true,
+      'delete': true,
+      'do': true,
+      'else': true,
+      'export': true,
+      'extends': true,
+      'finally': true,
+      'for': true,
+      'function': true,
+      'if': true,
+      'import': true,
+      'in': true,
+      'instanceof': true,
+      'let': true,
+      'new': true,
+      'return': true,
+      'super': true,
+      'switch': true,
+      'this': true,
+      'throw': true,
+      'try': true,
+      'typeof': true,
+      'var': true,
+      'void': true,
+      'while': true,
+      'with': true,
+      'yield': true,
+      'enum': true,
+      'await': true,
+      'int': true,
+      'byte': true,
+      'char': true,
+      'goto': true,
+      'long': true,
+      'final': true,
+      'float': true,
+      'short': true,
+      'double': true,
+      'native': true,
+      'throws': true,
+      'boolean': true,
+      'abstract': true,
+      'volatile': true,
+      'transient': true,
+      'synchronized': true
+    };
 
-    return (reserved.indexOf(str) > -1);
+    return (reserved[str] ? true : false);
```

Comment:
This will fail for e.g. `reserved['toString']` as that exists in the prototype chain. Use `hasOwnProperty` to check.